lasagne/tests/layers/test_conv.py

```diff
@@ -571,6 +589,25 @@ def test_with_nones(self, DummyInputLayer, input, kernel, output, kwargs):
         assert actual.shape == output.shape
         assert np.allclose(actual, output)
 
+    @pytest.mark.parametrize(
+        "input, kernel, output, kwargs", list(transp_conv2d_test_sets(True)))
+    def test_with_output_size(self, DummyInputLayer, input, kernel, output,
+                              kwargs):
```

Comment:
As I said (https://github.com/Lasagne/Lasagne/pull/745#pullrequestreview-1789166), I'd prefer to just add a few test cases to `transp_conv2d_test_sets`, so they're run by the existing `test_defaults` and `test_with_nones`. You don't need the `with_output` flag in `transp_conv2d_test_sets` then. Just add a few manual test cases at the end of the function, and make sure they include the `output_size` argument -- both with a fixed shape tuple, and with a shared variable (like T.shared(output.shape)). The idea of using the size of the ground truth output for the output_size is good, makes it easier.